From bc9790672fe7fa238e15628c87a64aa7cbe1f45f Mon Sep 17 00:00:00 2001
From: Raul Tambre <raul.tambre@gmail.com>
Date: Tue, 3 Jan 2017 14:37:44 +0200
Subject: [PATCH] Fix the option for turning off RTTI support in msvc format

"/GR-" must be explicitly specified, not using "/GR" is not enough with any
version of MSVC from this millennium.

See https://github.com/vslavik/bakefile/pull/85
---
 NEWS                         | 6 ++++++
 rules/makefile_defs_msvc.bkl | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index 2fb0b90..ab7358c 100644
--- a/NEWS
+++ b/NEWS
@@ -1,4 +1,10 @@
 
+Version 0.3.0
+-------------
+
+- msvc: Fix RTTI off flag (Raul Tambre)
+
+
 Version 0.2.9
 -------------
 
diff --git a/rules/makefile_defs_msvc.bkl b/rules/makefile_defs_msvc.bkl
index 6c46665..9823654 100644
--- a/rules/makefile_defs_msvc.bkl
+++ b/rules/makefile_defs_msvc.bkl
@@ -94,7 +94,7 @@
         <set var="__FLAG_WARNINGS_DEFAULT">$(__FLAG_MSVC_WARNINGS_DEFAULT)</set>
         <set var="__FLAG_WARNINGS_MAX">$(__FLAG_MSVC_WARNINGS_MAX)</set>
         <set var="__FLAG_RTTI_ON">/GR</set>
-        <set var="__FLAG_RTTI_OFF"></set>
+        <set var="__FLAG_RTTI_OFF">/GR-</set>
         <set var="__FLAG_EXCEPTIONS_ON">/EHsc</set>
         <set var="__FLAG_EXCEPTIONS_OFF"></set>
 
